provider/pkg/docs-gen/examples/image.md

```diff
@@ -137,3 +137,258 @@ public class App {
+        },
+    });
+
+    return new Dictionary<string, object?>
+    {
+        ["imageName"] = myAppImage.ImageName,
+    };
+});
+
+```
+```go
+package main
+
+import (
+	"fmt"
+
+	"github.com/pulumi/pulumi-aws/sdk/v5/go/aws/ecr"
+	"github.com/pulumi/pulumi-docker/sdk/v4/go/docker"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func main() {
+	pulumi.Run(func(ctx *pulumi.Context) error {
+		ecrRepository, err := ecr.NewRepository(ctx, "ecr-repository", &ecr.RepositoryArgs{
+			Name: pulumi.String("docker-repository"),
+		})
+		if err != nil {
+			return err
+		}
+		authToken := ecr.GetAuthorizationTokenOutput(ctx, ecr.GetAuthorizationTokenOutputArgs{
+			RegistryId: ecrRepository.RegistryId,
+		}, nil)
+		myAppImage, err := docker.NewImage(ctx, "my-app-image", &docker.ImageArgs{
+			Build: &docker.DockerBuildArgs{
+				Args: pulumi.StringMap{
+					"BUILDKIT_INLINE_CACHE": pulumi.String("1"),
+				},
+				CacheFrom: &docker.CacheFromArgs{
+					Images: pulumi.StringArray{
+						ecrRepository.RepositoryUrl.ApplyT(func(repositoryUrl string) (string, error) {
+							return fmt.Sprintf("%v:latest", repositoryUrl), nil
+						}).(pulumi.StringOutput),
+					},
+				},
+				Context:    pulumi.String("app/"),
+				Dockerfile: pulumi.String("Dockerfile"),
+			},
+			ImageName: ecrRepository.RepositoryUrl.ApplyT(func(repositoryUrl string) (string, error) {
+				return fmt.Sprintf("%v:latest", repositoryUrl), nil
+			}).(pulumi.StringOutput),
+			Registry: &docker.RegistryArgs{
+				Password: pulumi.ToSecret(authToken.ApplyT(func(authToken ecr.GetAuthorizationTokenResult) (string, error) {
+					return authToken.Password, nil
+				}).(pulumi.StringOutput)).(pulumi.StringOutput),
+				Server: ecrRepository.RepositoryUrl,
+			},
+		})
+		if err != nil {
+			return err
+		}
+		ctx.Export("imageName", myAppImage.ImageName)
+		return nil
+	})
+}
+```
+```yaml
+config: {}
+description: Docker image build using caching with AWS Elastic Container Registry
+name: image-caching-yaml
+outputs:
+    imageName: ${my-app-image.imageName}
+resources:
+    ecr-repository:
+        properties:
+            name: docker-repository
+        type: aws:ecr:Repository
+    my-app-image:
+        options:
+            version: v4.1.2
+        properties:
+            build:
+                args:
+                    BUILDKIT_INLINE_CACHE: "1"
+                cacheFrom:
+                    images:
+                        - ${ecr-repository.repositoryUrl}:latest
+                context: app/
+                dockerfile: Dockerfile
+            imageName: ${ecr-repository.repositoryUrl}:latest
+            registry:
+                password:
+                    fn::secret: ${authToken.password}
+                server: ${ecr-repository.repositoryUrl}
+        type: docker:Image
+runtime: yaml
+variables:
+    authToken:
+        fn::aws:ecr:getAuthorizationToken:
+            registryId: ${ecr-repository.registryId}
+```
+```java
+package generated_program;
```

Comment:
slightly offtopic: I think maybe we want to file something in pulumi-java or pulumi convert? not super up on current java but this feels like it could reduce a user's faith in the validity of the program. Is it copypastable with this package name?

Reply:
@iwahbe this issue belongs in pulumi-java, yeah?

Reply:
I think this is actually OK - Java code has to have a package name at the top of the file, like Go code does. Perhaps the only issue is that "generated_program" isn't canonical, and there's a better default?

Reply:
oh I definitely think this is OK for now - only that we should aim to improve the package name itself.